megamek/src/megamek/client/ui/swing/lobby/sorters/PlayerUnitRoleSorter.java

```diff
@@ -60,16 +62,10 @@ public int compare(final Entity a, final Entity b) {
             return t_a - t_b;
         } else if (!p_a.equals(p_b)) {
             return p_a.getName().compareTo(p_b.getName());
-        } else if ((ASConverter.canConvert(a)) && (ASConverter.canConvert(b))) {
-            FlexibleCalculationReport reportA = new FlexibleCalculationReport();
-            AlphaStrikeElement elementA = ASConverter.convert(a, false, reportA);
-            FlexibleCalculationReport reportB = new FlexibleCalculationReport();
-            AlphaStrikeElement elementB = ASConverter.convert(b, false, reportB);
-            String aRole = elementA.getRole().toString();
-            String bRole = elementB.getRole().toString();
-            return aRole.compareTo(bRole) * d;
         } else {
-            return 0;
+            String aRole = UnitRoleHandler.getRoleFor(a).toString();

```

Comment:
Can getRoleFor result in a null?

Reply:
it checks for null and returns UnitRole.UNDETERMINED


        if (null != role) {
            return role;
        }
        return UnitRole.UNDETERMINED;